### ISSUE TITLE
FlowRouter.getQueryParams() suggestion

### ISSUE THREAD

**Issue by Streemo** [meteorhacks/flow-router#220]
Hi again @arunoda ! First, thanks a bunch for all the work you do.

So, I was using your package, and I came up with a really ugly solution for a problem, and I figured others might like to see an official solution!

Frankly, `FlowRouter.watchPathChange()` is overkill for my purposes. It exhibits very strange behavior.  but `FlowRouter.getQueryParam('requiredArgument')` is too little. I need to watch my path for 4 or 5 query params, but not all 20. In fact, watching all 20 wouldn't even be bad. But `FlowRouter.watchPathChange()` comes with even extra baggage and is less predictable.

**Suggestion: `FlowRouter.getQueryParams()` called with no arguments, which returns the entire reactive queryParams object, or it can be given a list of params to watch** One could still call it with a string argument to obtain the old behavior.

This is a function I am using:

```
getQueryParamsReactive = function(list){
    //list is an array of query params I want to "watch".
	return _.reduce(list, function(memo,query){
		var param = FlowRouter.getQueryParam(query);
		if (param){
			memo[query] = param;
		}
		return memo;
	},{})
}
```

Basically, it watches the path for changes only in the desired query params I care about, but it's kind of ugly. 

Anyone else wanna see this feature? +1 this. 

Thanks :D!

**Comment by arunoda** [comment]
I think i'll keep this open, but I'm not going to implement this right now since there's a working solution for this. 
In Flow, we are not going to implement all the features we need, but layout the foundation. May be you can build a new package which expose this functionality.

**Comment by Streemo** [comment]
My solution resorts to calling FlowRouter.getqueryParam('param_k') for all N params - it just seems like there's a better way. i'll see if i cant find a good way to add onto FlowRouter object for a package!

**Comment by arunoda** [comment]
I think your solutions seems good. I can't think of another way.

**Comment by arunoda** [comment]
Closing this for now.
